packages/clay-drop-down/src/DropDownWithItems.tsx

```diff
@@ -615,78 +645,112 @@ export const ClayDropDownWithItems = ({
 
 	const Wrap = footerContent ? 'form' : React.Fragment;
 
-	return (
-		<ClayDropDown
-			active={internalActive}
-			alignmentByViewport={alignmentByViewport}
-			alignmentPosition={alignmentPosition}
-			className={className}
-			closeOnClickOutside={closeOnClickOutside}
-			containerElement={containerElement}
-			hasLeftSymbols={hasLeftSymbols}
-			hasRightSymbols={hasRightSymbols}
-			menuElementAttrs={menuElementAttrs}
-			menuHeight={menuHeight}
-			menuWidth={menuWidth}
-			offsetFn={offsetFn}
-			onActiveChange={setInternalActive}
-			renderMenuOnClick={renderMenuOnClick}
-			trigger={React.cloneElement(trigger, {
-				ref: (node: HTMLButtonElement) => {
-					if (node) {
-						triggerElementRef.current = node;
-						// Call the original ref, if any.
-						const {ref} = trigger;
-						if (typeof ref === 'function') {
-							ref(node);
+	const defaultActiveMenu = useId();
+
+	const windowWidth = window.innerWidth;
```

Comment:
You can use our `useIsMobileDevice` hook in `@clayui/shared` for this. One thing to note is that when the user resizes the screen this value does not change so we would have to add an event to update this value whenever it is resized. We can do this in the hook with a throttle if necessary.

Reply:
Since @pat270 is on PTO, I am working on this fix

Reply:
@matuzalemsteles, I've implemented the fix, but I ran into some issues due to the way hooks render. Since hooks can't be called inside an if statement and `useId()` relies on `useMemo`, I worked around it by adding a separate ID specifically for the drilldown.

Do you have any suggestions for a different approach?

Thanks!

Reply:
@ilzamcmed I'll take a look at this. The ideal is to have a unique ID per instance because we can have several components rendered on the page and this will have the same ID.